frostsnap_core/src/device.rs

```diff
@@ -69,36 +63,49 @@ impl FrostSigner {
         DeviceId::new(self.keypair().public_key())
     }
 
-    pub fn generate_nonces(
-        &mut self,
-        aux_rand: [u8; 32],
-        start: usize,
-        n: usize,
+    fn generate_nonces(
+        // this is always the device key for now but because of lifetimes issues it has to be passed in
```

Comment:
Is this still the case?

I tried changing to 
```
 fn generate_nonces(
        &self,
        start: u64,
)
```
and everything seems to compile, maybe i'm missing something!

Reply:
Good catch. Thanks!